### DISCUSSION TITLE
How to use until-found

### DISCUSSION THREAD

**Discussion by NxP4Code** [icloud-photos-downloader/icloud_photos_downloader#477] [Q&A]
I am trying to understand how to use the until-found flag. Can someone provide an explanation of the until-found flag?
## Summary
[NOTE]: # ( Provide a brief overview of what the new feature is all about )

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )

**Comment by krzyk** [comment]
Yeah, the description is quite cryptic.

**Comment by AlexanderSch90** [comment]
+1

**Comment by AndreyNikiforov** [comment]
The intent is to reduce synchronization time for periodic runs.

`icloudpd`:
- looks for photos in iCloud from latest to oldest
- checks if photo exists on the local drive
- downloads if photo does not exist locally
- if photo exists it moves to checking next photo
- however, if the number of consecutive photos that already exist on local drive matches or greater than the number supplied to `--until-found` parameter, then `icloudpd` stops the scan.

**Reply by vadalus911** [reply]
my experience of testing this is that the first point does not seem to work for me. It does not start from the most recent files and so finds matches (which are old previously downloaded files) and stops after the number X specified. Its quite clear to see this in the logs as the downloaded files are written to a dir structure which has year / month etc..

**Reply by AndreyNikiforov** [reply]
`until-found` uses date when photos were added to collection, not the date when photos were taken. Folder structure is built from date when photos were taken. If you recently uploaded old photos to iCloud, they will be downloaded and put in old folder structure.

**Reply by georgeguimaraes** [reply]
That's a great feature, thanks for explaining.

**Reply by toyo2333** [reply]
> `until-found` uses date when photos were added to collection, not the date when photos were taken. Folder structure is built from date when photos were taken. If you recently uploaded old photos to iCloud, they will be downloaded and put in old folder structure.`until-found` 使用照片添加到收藏的日期，而不是照片拍摄的日期。文件夹结构是根据照片拍摄日期建立的。如果你最近将旧照片上传到 iCloud，它们将被下载并放入旧的文件夹结构中。

that is a smart move
